### PR TITLE
Add customer update endpoint

### PR DESCRIPTION
**Pull request by davidmuzi**
### What is this

Fix: #272 

Adds support for the update customer endpoint.  Removes pointless customer update unit test.

@bgulanowski

### CODE REVIEW

File: Mobile Buy SDK/Mobile Buy SDK Tests/BUYClient+CustomerTests.m [hunk 1]
```diff
@@ -197,29 +197,6 @@ - (void)testCustomerOrderWithID
 	[self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {}];
 }
 
-#pragma mark - Update -
-
-- (void)testCustomerUpdate
```
Comment (davidmuzi) [inline Mobile Buy SDK/Mobile Buy SDK Tests/BUYClient+CustomerTests.m]: Is there a point in mocking an `update` method?

Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK Tests/BUYClient+CustomerTests.m]: To make sure that the callback is invoked, and with the right kinds of objects?

Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK Tests/BUYClient+CustomerTests.m]: When running this against Shopify, shouldn't we check, for example, that changing the default address does the right thing? If we had two, we could alternate, although then the test wouldn't be stateless (unless we changed it and then changed it back).

**Comment by bgulanowski** [comment]
Lookin' good! 🎩

**Pull request merged**